posthog/management/commands/add_question_ids_to_surveys.py

```diff
@@ -0,0 +1,164 @@
+            action="store_true",
+            help="Show detailed information about the process",
+        )
+
+    def handle(self, *args, **options):
+        batch_size = options["batch_size"]
+        really_run = options["really_run"]
+        verbose = options["verbose"]
+
+        if not really_run:
+            self.stdout.write(self.style.WARNING("Running in dry-run mode - no changes will be made"))
+            self.stdout.write(self.style.WARNING("Use --really-run to actually make changes"))
+
+        self.stdout.write(
+            self.style.WARNING(
+                "Note: This command may not be necessary for new surveys as IDs are automatically added via pre_save signal"
+            )
+        )
+        self.stdout.write(
+            self.style.WARNING(
+                "It's primarily useful for older surveys created before the pre_save signal was implemented"
+            )
+        )
+
+        self.add_question_ids_to_surveys(batch_size=batch_size, really_run=really_run, verbose=verbose)
+
+    def add_question_ids_to_surveys(self, batch_size: int, really_run: bool, verbose: bool) -> None:
+        """
+        Add unique IDs to each question in existing surveys.
+        This ensures all questions have an ID for tracking responses.
+
+        Processes surveys in small batches to minimize memory usage.
+        Uses proper row locking to prevent race conditions.
+        """
+        total_surveys = Survey.objects.filter(questions__isnull=False).count()
+        total_modified = 0
+        total_questions_updated = 0
+        skipped_surveys = 0
+
+        self.stdout.write(f"Found {total_surveys} surveys to process")
+
+        # Process in batches to minimize memory usage
+        offset = 0
+
+        while True:
+            # Process one batch at a time
+            with transaction.atomic():
+                # Get a batch of surveys WITH ROW LOCKING
+                # This ensures no one else can modify these rows while we're processing them
+                surveys = list(
+                    Survey.objects.filter(questions__isnull=False)
+                    .order_by("pk")
+                    .select_for_update(skip_locked=True)  # Add row locking
+                    .only("id", "questions")[
+                        # Only fetch fields we need
+                        offset : offset + batch_size
+                    ]
+                )
+
+                if not surveys:
+                    # If we got no surveys, it could be because they're all locked
+                    # or because we've processed all of them
+                    if skipped_surveys > 0:
+                        # If we've skipped surveys before, try again from the beginning
+                        # to catch any that were previously locked
+                        offset = 0
+                        skipped_surveys = 0
+                        self.stdout.write("Restarting from the beginning to process previously locked surveys")
```

Comment:
can this potentially become an endless loop?

Reply:
> it can't

the code path allowed that with certain conditions, such as something deadlocked, so it can, so yes, better to just rerun rather than killing commands manually if needed